UIKit/S2MFoldAnimator.m

```diff
@@ -0,0 +1,417 @@
+                             leadingViewFold.layer.position = [self finalPositionInView:view
+                                                                                leading:YES
+                                                                             foldOffset:viewOffset
+                                                                            foldMeasure:viewFoldMeasure];
+                             [self applyTransformForView:leadingViewFold leading:YES initial:NO];
+                             [leadingViewFold.subviews[1] setAlpha:(self.unfolding) ? 0.0 : 1.0];
+                             
+                             UIView *trailingViewFold =  viewFolds[i*2+1];
+                             trailingViewFold.layer.position = [self finalPositionInView:view
+                                                                                 leading:NO
+                                                                              foldOffset:viewOffset
+                                                                             foldMeasure:viewFoldMeasure];
+                             [self applyTransformForView:trailingViewFold leading:NO initial:NO];
+                             [trailingViewFold.subviews[1] setAlpha:(self.unfolding) ? 0.0 : 1.0];
+                         }
+                     }  completion:^(BOOL finished) {
+                         [UIView performWithoutAnimation:^{
+                             // restore the to- and from- to the initial location
+                             if (self.unfolding) {
+                                 // put back toView frame to expected location
+                                 view.frame = CGRectMake(viewOrigin.x, viewOrigin.y, viewSize.width, viewSize.height);
+                             }else{
+                                 if (self.direction == S2MFoldAnimatorDirectionRightToLeft) {
+                                     view.frame = CGRectMake(viewOrigin.x, viewOrigin.y, 0, viewSize.height);
+                                 }else if (self.direction == S2MFoldAnimatorDirectionLeftToRight) {
+                                     view.frame = CGRectMake(viewOrigin.x + viewSize.width, viewOrigin.y, 0, viewSize.height);
+                                 }else{
+                                     // Disable this in order not to break constraints
+                                 }
+                             }
+                             view.hidden = NO;
+                             // remove the snapshot views
+                             for (UIView *view in viewFolds) {
+                                 [view removeFromSuperview];
+                             }
+                         }];
+                         
+                         if(completion){
+                             completion(finished);
+                         }
+                     }];
+}
+
+#pragma mark - Private
+
```

Comment:
@joernito it depends on the direction of the fold animator object. any suggestion for naming?